server/graphql_schemas/event/schema.py

```diff
@@ -23,43 +25,82 @@ class Input:
         date = graphene.String(required=True)
         time = graphene.String(required=False)
         featured_image = graphene.String(required=False)
-        social_event_id = graphene.Int(required=False)
+        social_event_id = graphene.String(required=False)
 
     new_event = graphene.Field(EventNode)
 
     @classmethod
     def mutate_and_get_payload(cls, root, info, **input):
-        user = info.context.user
         social_event_id = input.get('social_event_id')
-        social_event = Category.objects.get(id=int(social_event_id))
-        andela_user_profile = AndelaUserProfile.objects.get(user_id=user.id)
-        new_event = Event(
-          title=input.get('title'),
-          description=input.get('description'),
-          venue=input.get('venue'),
-          date=input.get('date'),
-          time=input.get('time'),
-          featured_image=input.get('featured_image'),
-          creator=andela_user_profile,
-          social_event=social_event
-        )
-        new_event.save()
+        try:
+            social_event = Category.objects.get(
+                id=from_global_id(social_event_id)[1])
+            user_profile = AndelaUserProfile.objects.get(
+                user=info.context.user
+                )
+            new_event = Event(
+                title=input.get('title'),
+                description=input.get('description'),
+                venue=input.get('venue'),
+                date=input.get('date'),
+                time=input.get('time'),
+                featured_image=input.get('featured_image'),
+                creator=user_profile,
+                social_event=social_event
+            )
+            new_event.save()
+
+        except ValueError as e:
+            raise GraphQLError("An Error occured. \n{}".format(e))
 
         return cls(new_event=new_event)
 
 
+class UpdateEvent(relay.ClientIDMutation):
+
+    class Input:
+        title = graphene.String()
+        description = graphene.String()
+        venue = graphene.String()
+        date = graphene.String()
+        time = graphene.String()
+        featured_image = graphene.String()
+        social_event_id = graphene.Int()
```

Comment:
well done @parseendavid But line 68 should be graphene.String() like the line  28.

Reply:
on it